### PR TITLE
Issue399

### PR DESCRIPTION
**Pull request by yankev**
Changed the parameters for all the functions from taking on `linkText` and `showLink` to take on `config` (which is a dictionary of possible configurations).

### CODE REVIEW

**Comment by yankev** [comment]
@cldougl @chriddyp

File: plotly/offline/offline.py [hunk 1]
```diff
@@ -97,22 +99,44 @@ def _plot_html(figure_or_data, show_link, link_text,
...
+        'linkText',
+        'showSources',
+        'displayModeBar',
+        'modeBarButtonsToRemove',
+        'modeBarButtonsToAdd',
+        'modeBarButtons',
+        'displaylogo',
+        'plotGlPixelRatio',
+        'setBackground',
+        'topojsonURL')
```
Comment (etpinard) [inline plotly/offline/offline.py]: What if we add another config option in plotly.js? This list can't stay in-sync.

Comment (etpinard) [inline plotly/offline/offline.py]: @cpsievert got this right, by not validating the config options: https://github.com/ropensci/plotly/pull/446/files

At some point, I want to make the config option part of the plot schema, where they could be use to validate user input from the python api. Before that happens though, I would much prefer not validating the config options.

**Comment by jackparmer** [comment]
Related: https://github.com/plotly/plotly.py/issues/399

File: plotly/offline/offline.py [hunk 2]
```diff
@@ -137,7 +161,8 @@ def _plot_html(figure_or_data, show_link, link_text,
     return plotly_html_div, plotdivid, width, height
 
 
-def iplot(figure_or_data, show_link=True, link_text='Export to plot.ly',
+def iplot(figure_or_data,
+          config={'showLink': True, 'linkText': 'Export to plot.ly'},
```
Comment (chriddyp) [inline plotly/offline/offline.py]: we try really hard not to break backwards incompatibility. if we remove `show_link` and `link_text` then our user's code could break when they upgrade plotly. in this case, we might want to do something like:

```
iplot(figure_or_data, show_link=True, link_text='Export to plot.ly', **config_options)
```

File: plotly/offline/offline.py [hunk 3]
```diff
@@ -152,10 +177,12 @@ def iplot(figure_or_data, show_link=True, link_text='Export to plot.ly',
...
-    show_link (default=True) -- display a link in the bottom-right corner of
-                                of the chart that will export the chart to
-                                Plotly Cloud or Plotly Enterprise
-    link_text (default='Export to plot.ly') -- the text of export link
+    config -- a dictionary with configuration attributes for the plot,
+        default contains the following two elements:
+        showLink (default=True) -- display a link in the bottom-right corner of
+                                    of the chart that will export the chart to
+                                    Plotly Cloud or Plotly Enterprise
+        linkText (default='Export to plot.ly') -- the text of export link
```
Comment (chriddyp) [inline plotly/offline/offline.py]: we'll want to describe that these are `plotly.js` configuration options and provide a link to our documentation: https://plot.ly/javascript/configuration-options/

File: contributing.md [hunk 4]
```diff
@@ -98,7 +98,7 @@ To install the optional dependencies:
 pip install -r optional-requirements.txt
 ```
 
-##Testing
+##Testing
```
Comment (chriddyp) [inline contributing.md]: I'd set up your editor to trim trailing white spaces on save

File: plotly/offline/offline.py [hunk 5]
```diff
@@ -9,9 +9,11 @@
 import os
 import uuid
 import warnings
-from pkg_resources import resource_string
 import webbrowser
 
+from pkg_resources import resource_string
+
+
```
Comment (chriddyp) [inline plotly/offline/offline.py]: it looks like this got accidentally committed - GitHub Desktop makes it easy to not commit and discard these types of changes

File: plotly/offline/offline.py (hunk 3 above)
Comment (yankev) [inline plotly/offline/offline.py]: Ok.  Something to note though is that the keywords for show_link ad link_text aren't in sync with the ones on the `plotly.js config options`, which could lead to some confusion.

**Comment by yankev** [comment]
Made the changes in a new branch, and create a new pull request here:
[https://github.com/plotly/plotly.py/pull/410](https://github.com/plotly/plotly.py/pull/410)

**Pull request closed without merging**